internal/lightcone/harmony/memoriesofthepast/memoriesofthepast.go

```diff
@@ -0,0 +1,80 @@
+package memoriesofthepast
+
+import (
+	"github.com/simimpact/srsim/pkg/engine"
+	"github.com/simimpact/srsim/pkg/engine/equip/lightcone"
+	"github.com/simimpact/srsim/pkg/engine/event"
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/modifier"
+	"github.com/simimpact/srsim/pkg/engine/prop"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+const (
+	memories = "memories-of-the-past"
+)
+
+type state struct {
+	onCooldown bool
+	energyAmt  float64
+}
+
+// Increases the wearer's Break Effect by 28%. When the wearer attacks,
+// additionally regenerates 4 Energy. This effect can only be triggered 1 time per turn.
+
+func init() {
+	lightcone.Register(key.MemoriesofthePast, lightcone.Config{
+		CreatePassive: Create,
+		Rarity:        4,
+		Path:          model.Path_HARMONY,
+		Promotions:    promotions,
+	})
+	modifier.Register(memories, modifier.Config{
+		Listeners: modifier.Listeners{
+			OnAfterAttack: addEnergy,
+			// DM uses OnListenTurnEnd. try to imitate with normal and skill action check.
+			OnAfterAction: resetCooldown,
```

Comment:
`OnAfterAction` is not equivalent to `OnListenTurnEnd`. Anything that starts with `OnListen` in the datamine means it triggers on **any** event rather than events associated with the character (which is what `OnAfterAction` would be). Since Memories of the Past uses `OnListenTurnEnd`, this means that the energy gain will reset on **any** turn and not just the turn of the equipped character (so can get energy from any ults or insert attacks that occur outside of the character's turn)

The best way to do this would be with event subscriptions (event subscriptions are our equivalent of `OnListen*`). I would say implement similar to "Time Waits for No One" which uses subscriptions for the logic. That is:
* `TurnEnd` subscription which resets the cooldown
* `AttackEnd` subscription that filters for only attacks performed by the LC owner, which on those attacks (if not on CD) give energy and give CD.
* 
You could also have the CD stored as a modifier for better traceability in the sim